VM image build - system dependency for R package prevents complete build

## :white_check_mark: Checklist

<!--
Before reporting a problem please check the following. Replace the empty checkboxes [ ] below with checked ones [x] accordingly.
-->

- [x] I have searched open and closed issues for duplicates.
- [x] This is a problem observed when **deploying** a Data Safe Haven.
- [x] I can reproduce this with the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :computer: System information

- **Operating System:** Mac OS
- **Data Safe Haven version:** Develop

## :cactus: Powershell module versions

<!-- Paste the output of deployment/CheckRequirements.ps1 here -->

Not relevant

## :no_entry_sign: Describe the problem

<!--
What happens now and (if relevant) an explanation of why this is incorrect

Optionally include specific tasks (if any) in the order in which they need to be done.
- [ ] Task 1
- [ ] Task 2
- [ ] Task 3

Include a clear and concise description of what the behaviour should be when this issue is closed if this is not obvious.
-->

The compilation of the package `igraph` fails due to a missing system dependency, `glpk`

## :deciduous_tree: Log messages

<details>
<summary>Relevant log messages</summary>

```none
srdbuildadmin@CandidateSecureResearchDesktop-Ubuntu2004-202402011530:~$ /opt/monitoring/analyse_build.py 
[2024-02-01 15:31:15: SUCCESS] Build started: 
[2024-02-01 15:32:27: SUCCESS] Running cloud-init modules: 1m12s
[2024-02-01 16:01:06: SUCCESS] Preparing system for installation: 28m38s
[2024-02-01 16:03:11: SUCCESS] Installing additional deb/source packages: 2m5s
[2024-02-01 16:07:00: SUCCESS] Updating ClamAV database: 3m49s
[2024-02-01 16:08:39: SUCCESS] Installing Spark/Hadoop: 1m39s
[2024-02-01 16:21:11: SUCCESS] Installing Julia and packages: 12m32s
[2024-02-01 16:36:14: SUCCESS] Installing Python (3.9.18) and packages: 15m3s
[2024-02-01 16:48:31: SUCCESS] Installing Python (3.10.13) and packages: 12m17s
[2024-02-01 17:00:17: SUCCESS] Installing Python (3.11.7) and packages: 11m46s
[2024-02-01 17:00:25: SUCCESS] Configuring pyenv: 8s
[2024-02-01 17:17:56: FAIL   ] Installing R packages: 17m31s
[2024-02-01 17:17:57: FAIL   ] Finished build: 1h46m42s
[2024-02-05 12:28:10: RUNNING] Installing/updating Ubuntu packages: 3d19h10m12s
[2024-02-01 17:17:56: INFO   ] Memory available: 17 GB
[2024-02-01 17:17:56: INFO   ] ..... mean usage:  10.54% =>  1.8 GB
[2024-02-01 17:17:56: INFO   ] ...... min usage:   5.30% =>  0.9 GB
[2024-02-01 17:17:56: INFO   ] ...... max usage:  31.87% =>  5.3 GB
[2024-02-01 17:17:56: INFO   ] CPU available: 8 cores
[2024-02-01 17:17:56: INFO   ] ..... mean usage:  38.25% =>  3.1 cores
[2024-02-01 17:17:56: INFO   ] ...... min usage:   0.10% =>  0.0 cores
[2024-02-01 17:17:56: INFO   ] ...... max usage: 100.00% =>  8.0 cores
```

```none
In file included from vendor/cigraph/src/community/optimal_modularity.c:32:
vendor/cigraph/src/internal/glpk_support.h:39:10: fatal error: glpk.h: No such file or directory
   39 | #include <glpk.h>
      |          ^~~~~~~~
compilation terminated.
make[1]: *** [/usr/lib/R/etc/Makeconf:191: vendor/cigraph/src/community/optimal_modularity.o] Error 1
make[1]: Leaving directory '/tmp/Rtmp9mkHx8/R.INSTALL278003285c5eb/igraph/src'
ERROR: compilation failed for package ‘igraph’
* removing ‘/usr/local/lib/R/site-library/igraph’
cat: networkD3.out: No such file or directory
cat: DataExplorer.out: No such file or directory
```
</details>

Adding `libgplk-dev` to the apt list will install the relevant system library.

## :recycle: To reproduce

<!--
How can the current behaviour be reproduced? Include some steps to reproduce the problem.

Example:
- Deploy an SRE with option X set to Y
- Attempt to perform action Z
- Observe the previously mentioned failure/error message
-->

Build a VM image

--- a/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2004.mustache.yaml
+++ b/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2004.mustache.yaml
@@ -407,7 +407,7 @@ runcmd:
 
   # Remove Linux omsagent before generalising the image
   - echo ">=== $(date +%s) Removing Linux omsagent ===<"
-  - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh
+  - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/c7e1a8f923844f989c8aa7957cee0e89708d821f/installer/scripts/onboard_agent.sh
   - sh onboard_agent.sh --purge || die "Could not remove Linux omsagent!";
   - rm onboard_agent.sh
 

--- a/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2004.mustache.yaml
+++ b/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2004.mustache.yaml
@@ -404,7 +404,7 @@ runcmd:
   - sed -i "s|^export PATH=.*|export PATH=${PATH}|" /etc/bash.bashrc
   # Run installation tests
   - python3 /opt/build/check_installation.py || die "Installation check failed!";
-  
+
   # Remove Linux omsagent before generalising the image
   - echo ">=== $(date +%s) Removing Linux omsagent ===<"
   - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh

--- a/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2004.mustache.yaml
+++ b/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2004.mustache.yaml
@@ -407,6 +407,7 @@ runcmd:
 
   # Remove Linux omsagent before generalising the image
   - echo ">=== $(date +%s) Removing Linux omsagent ===<"
+  # We use the commit corresponding to a particular tag (currently 1.17.2-0) so that we can audit what the script is doing
   - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/c7e1a8f923844f989c8aa7957cee0e89708d821f/installer/scripts/onboard_agent.sh
   - sh onboard_agent.sh --purge || die "Could not remove Linux omsagent!";
   - rm onboard_agent.sh

--- a/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2004.mustache.yaml
+++ b/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2004.mustache.yaml
@@ -404,6 +404,12 @@ runcmd:
   - sed -i "s|^export PATH=.*|export PATH=${PATH}|" /etc/bash.bashrc
   # Run installation tests
   - python3 /opt/build/check_installation.py || die "Installation check failed!";
+  
+  # Remove Linux omsagent before generalising the image
+  - echo ">=== $(date +%s) Removing Linux omsagent ===<"
+  - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh
+  - sh onboard_agent.sh --purge || die "Could not remove Linux omsagent!";
+  - rm onboard_agent.sh
 
 final_message: "System setup through cloud-init is finished. Configuration took $UPTIME seconds"
 

--- a/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2204.mustache.yaml
+++ b/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2204.mustache.yaml
@@ -410,13 +410,13 @@ runcmd:
   - sed -i "s|^export PATH=.*|export PATH=${PATH}|" /etc/bash.bashrc
   # Run installation tests
   - python3 /opt/build/check_installation.py || die "Installation check failed!";
-  
+
   # Remove Linux omsagent before generalising the image
   - echo ">=== $(date +%s) Removing Linux omsagent ===<"
   - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh
   - sh onboard_agent.sh --purge || die "Could not remove Linux omsagent!";
   - rm onboard_agent.sh
-  
+
 final_message: "System setup through cloud-init is finished. Configuration took $UPTIME seconds"
 
 # Shutdown at the end of the job to save on running costs

--- a/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2204.mustache.yaml
+++ b/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2204.mustache.yaml
@@ -413,7 +413,7 @@ runcmd:
 
   # Remove Linux omsagent before generalising the image
   - echo ">=== $(date +%s) Removing Linux omsagent ===<"
-  - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh
+  - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/c7e1a8f923844f989c8aa7957cee0e89708d821f/installer/scripts/onboard_agent.sh
   - sh onboard_agent.sh --purge || die "Could not remove Linux omsagent!";
   - rm onboard_agent.sh
 

--- a/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2204.mustache.yaml
+++ b/deployment/secure_research_desktop/cloud_init/cloud-init-buildimage-ubuntu-2204.mustache.yaml
@@ -410,7 +410,13 @@ runcmd:
   - sed -i "s|^export PATH=.*|export PATH=${PATH}|" /etc/bash.bashrc
   # Run installation tests
   - python3 /opt/build/check_installation.py || die "Installation check failed!";
-
+  
+  # Remove Linux omsagent before generalising the image
+  - echo ">=== $(date +%s) Removing Linux omsagent ===<"
+  - wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh
+  - sh onboard_agent.sh --purge || die "Could not remove Linux omsagent!";
+  - rm onboard_agent.sh
+  
 final_message: "System setup through cloud-init is finished. Configuration took $UPTIME seconds"
 
 # Shutdown at the end of the job to save on running costs

--- a/deployment/secure_research_desktop/packages/packages-apt.list
+++ b/deployment/secure_research_desktop/packages/packages-apt.list
@@ -46,6 +46,7 @@ libffi-dev
 libfftw3-dev
 libfribidi-dev
 libgdal-dev
+libglpk-dev
 libgmp-dev
 libgsl-dev
 libgtk2.0-dev

--- a/environment_configs/sre_bluet1guac_core_config.json
+++ b/environment_configs/sre_bluet1guac_core_config.json
@@ -8,7 +8,7 @@
     "outboundInternetAccess": "default",
     "computeVmImage": {
         "type": "Ubuntu",
-        "version": "20.04.2024011200"
+        "version": "20.04.2024020500"
     },
     "remoteDesktopProvider": "ApacheGuacamole",
     "dataAdminIpAddresses": ["193.60.220.253"],

--- a/environment_configs/sre_bluet2guac_core_config.json
+++ b/environment_configs/sre_bluet2guac_core_config.json
@@ -8,7 +8,7 @@
     "outboundInternetAccess": "default",
     "computeVmImage": {
         "type": "Ubuntu",
-        "version": "20.04.2024011200"
+        "version": "20.04.2024020500"
     },
     "remoteDesktopProvider": "ApacheGuacamole",
     "dataAdminIpAddresses": ["193.60.220.253"],

--- a/environment_configs/sre_bluet3guac_core_config.json
+++ b/environment_configs/sre_bluet3guac_core_config.json
@@ -8,7 +8,7 @@
     "outboundInternetAccess": "default",
     "computeVmImage": {
         "type": "Ubuntu",
-        "version": "20.04.2024011200"
+        "version": "20.04.2024020500"
     },
     "remoteDesktopProvider": "ApacheGuacamole",
     "dataAdminIpAddresses": ["193.60.220.240"],

--- a/environment_configs/sre_greent1guac_core_config.json
+++ b/environment_configs/sre_greent1guac_core_config.json
@@ -8,7 +8,7 @@
     "outboundInternetAccess": "default",
     "computeVmImage": {
         "type": "Ubuntu",
-        "version": "20.04.2024011200"
+        "version": "20.04.2024020500"
     },
     "remoteDesktopProvider": "ApacheGuacamole",
     "dataAdminIpAddresses": ["193.60.220.253"],

--- a/environment_configs/sre_greent2guac_core_config.json
+++ b/environment_configs/sre_greent2guac_core_config.json
@@ -8,7 +8,7 @@
     "outboundInternetAccess": "default",
     "computeVmImage": {
         "type": "Ubuntu",
-        "version": "20.04.2024011200"
+        "version": "20.04.2024020500"
     },
     "remoteDesktopProvider": "ApacheGuacamole",
     "dataAdminIpAddresses": ["193.60.220.253"],

--- a/environment_configs/sre_greent3guac_core_config.json
+++ b/environment_configs/sre_greent3guac_core_config.json
@@ -8,7 +8,7 @@
     "outboundInternetAccess": "default",
     "computeVmImage": {
         "type": "Ubuntu",
-        "version": "20.04.2024011200"
+        "version": "20.04.2024020500"
     },
     "remoteDesktopProvider": "ApacheGuacamole",
     "dataAdminIpAddresses": ["193.60.220.240"],

--- a/tests/resources/sre_bluet1guac_full_config.json
+++ b/tests/resources/sre_bluet1guac_full_config.json
@@ -1291,7 +1291,7 @@
       "rg": "RG_SHM_BLUE_SRE_T1GUAC_COMPUTE",
       "vmImage": {
         "type": "Ubuntu",
-        "version": "20.04.2024011200"
+        "version": "20.04.2024020500"
       },
       "vmSizeDefault": "Standard_D2s_v3"
     },

--- a/tests/resources/sre_bluet3guac_full_config.json
+++ b/tests/resources/sre_bluet3guac_full_config.json
@@ -1291,7 +1291,7 @@
       "rg": "RG_SHM_BLUE_SRE_T3GUAC_COMPUTE",
       "vmImage": {
         "type": "Ubuntu",
-        "version": "20.04.2024011200"
+        "version": "20.04.2024020500"
       },
       "vmSizeDefault": "Standard_D2s_v3"
     },

--- a/tests/resources/sre_greent2guac_full_config.json
+++ b/tests/resources/sre_greent2guac_full_config.json
@@ -1344,7 +1344,7 @@
       "rg": "RG_SHM_GREEN_SRE_T2GUAC_COMPUTE",
       "vmImage": {
         "type": "Ubuntu",
-        "version": "20.04.2024011200"
+        "version": "20.04.2024020500"
       },
       "vmSizeDefault": "Standard_D2s_v3"
     },

